File: Foody/app/src/main/java/prateek_gupta/foody/ui/fragments/overview/OverviewFragment.java
package prateek_gupta.foody.ui.fragments.overview;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.fragment.app.Fragment;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import org.jsoup.Jsoup;

import coil.Coil;
import coil.ImageLoader;
import coil.request.ImageRequest;
import prateek_gupta.foody.R;
import prateek_gupta.foody.bindingsAdapters.RecipesRowBinding;
import prateek_gupta.foody.databinding.FragmentOverviewBinding;
import prateek_gupta.foody.models.Result;
import prateek_gupta.foody.util.Constants;


public class OverviewFragment extends Fragment {

    FragmentOverviewBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding= FragmentOverviewBinding.inflate(inflater, container, false);

        if (getArguments() != null) {
            Result myBundle=getArguments().getParcelable(Constants.RECIPE_RESULT_KEY);

            ImageLoader imageLoader = Coil.imageLoader(binding.mainImageView.getContext());
            Log.d("OverviewFragment", "onCreateView: url : "+myBundle.image);
            ImageRequest request = new ImageRequest.Builder(binding.getRoot().getContext())
                    .data(myBundle.image)
                    .target(binding.mainImageView)
                    .build();
            imageLoader.enqueue(request);

            binding.titleTextView.setText(myBundle.title);

            binding.likesTextView.setText(String.valueOf(myBundle.aggregateLikes));

            binding.timeTextView.setText(String.valueOf(myBundle.readyInMinutes));

            RecipesRowBinding.parseHtml(binding.summaryTextView,myBundle.summary);

            updateColors(myBundle.vegetarian, binding.vegetarianTextView, binding.vegetarianImageView);
            updateColors(myBundle.vegan, binding.veganTextView, binding.veganImageView);
            updateColors(myBundle.cheap, binding.cheapTextView, binding.cheapImageView);
            updateColors(myBundle.dairyFree, binding.dairyFreeTextView, binding.dairyFreeImageView);
            updateColors(myBundle.glutenFree, binding.glutenFreeTextView, binding.glutenFreeImageView);
            updateColors(myBundle.veryHealthy, binding.healthyTextView, binding.healthyImageView);
        }

        return binding.getRoot();
    }

    void updateColors(Boolean stateIsOn , TextView textView, ImageView imageView) {
        if (stateIsOn) {
            imageView.setColorFilter(ContextCompat.getColor(requireContext(),R.color.green));
            textView.setTextColor(ContextCompat.getColor(requireContext(), R.color.green));
        }
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding=null;
    }
}

File: Foody/app/src/main/java/prateek_gupta/foody/ui/fragments/overview/OverviewFragment.kt
package prateek_gupta.foody.ui.fragments.overview

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.fragment.app.Fragment
import coil.load
import org.jsoup.Jsoup
import prateek_gupta.foody.R
import prateek_gupta.foody.databinding.FragmentOverviewBinding
import prateek_gupta.foody.models.Result
import prateek_gupta.foody.util.Constants

class OverviewFragment : Fragment() {

    private var _binding:FragmentOverviewBinding?=null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        _binding = FragmentOverviewBinding.inflate(inflater, container, false)

        val args = arguments
        val myBundle = args?.getParcelable<Result>(Constants.RECIPE_RESULT_KEY) as Result

        binding.mainImageView.load(myBundle.image)
        binding.titleTextView.text = myBundle.title
        binding.likesTextView.text = myBundle.aggregateLikes.toString()
        binding.timeTextView.text = myBundle.readyInMinutes.toString()
        myBundle.summary.let {
            val summary = Jsoup.parse(it).text()
            binding.summaryTextView.text = summary
        }


        if(myBundle.vegetarian){
            binding.vegetarianImageView.setColorFilter(ContextCompat.getColor(requireContext(), R.color.green))
            binding.vegetarianTextView.setTextColor(ContextCompat.getColor(requireContext(), R.color.green))
        }

        if(myBundle.vegan){
            binding.veganImageView.setColorFilter(ContextCompat.getColor(requireContext(), R.color.green))
            binding.veganTextView.setTextColor(ContextCompat.getColor(requireContext(), R.color.green))
        }

        if(myBundle.glutenFree){
            binding.glutenFreeImageView.setColorFilter(ContextCompat.getColor(requireContext(), R.color.green))
            binding.glutenFreeTextView.setTextColor(ContextCompat.getColor(requireContext(), R.color.green))
        }

        if(myBundle.dairyFree){
            binding.dairyFreeImageView.setColorFilter(ContextCompat.getColor(requireContext(), R.color.green))
            binding.dairyFreeTextView.setTextColor(ContextCompat.getColor(requireContext(), R.color.green))
        }

        if(myBundle.veryHealthy){
            binding.healthyImageView.setColorFilter(ContextCompat.getColor(requireContext(), R.color.green))
            binding.healthyTextView.setTextColor(ContextCompat.getColor(requireContext(), R.color.green))
        }

        if(myBundle.cheap){
            binding.cheapImageView.setColorFilter(ContextCompat.getColor(requireContext(), R.color.green))
            binding.cheapTextView.setTextColor(ContextCompat.getColor(requireContext(), R.color.green))
        }

        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding=null
    }
}